hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -295,11 +295,15 @@ private void handleExecuteResponse(
         SqlExecuteCodec.ResponseParameters response = SqlExecuteCodec.decodeResponse(message);
         SqlError sqlError = response.error;
         if (sqlError != null) {
+            Throwable cause = null;
```

Comment:
@hazelcast/sql Could you confirm that you are ok with this? Alternatively we could add a separate field to `HazelcastSqlException`.

Reply:
this is a bit ugly (textual stacktrace as message) but should do the job. I would only use RuntimeException here, but it is more a matter of taste.